kernel/src/ipc/semaphore/system_v/sem.rs

```diff
@@ -46,22 +47,15 @@ pub enum Status {
     Removed = 2,
 }
 
-struct AtomicStatus(AtomicU16);
-
-impl AtomicStatus {
-    fn new(status: Status) -> Self {
-        Self(AtomicU16::new(status as u16))
-    }
-
-    fn status(&self) -> Status {
-        Status::try_from(self.0.load(Ordering::Relaxed)).unwrap()
-    }
-
-    fn set_status(&self, status: Status) {
-        self.0.store(status as u16, Ordering::Relaxed);
+impl From<Status> for u16 {
+    fn from(value: Status) -> Self {
+        value as u16
     }
 }
 
+#[atomic_version_of_integer_like_type(Status, try_from = true)]
+struct AtomicStatus(AtomicU16);
```

Comment:
I still think defining the macro as the function-style one is better.

```rust
impl_atomic_version_of_integer_like_type!(Status, try_from = true, {
     struct AtomicStatus(AtomicU16);
});
```

The function style, especially with its name starting with the verb `impl` (or `define`), makes it clear that the macro will only add implementation blocks to the type, without modifying the struct definition. On the other hand, it is less clear what an attribute-style macro would add to a type struct. In theory, it could transform the type definition into another form (e.g., adding some hidden fields). Macros are magical; so we should make our best try to aid the user/reader to understand and reason about it.

Reply:
This is only a suggestion. Using attribute macro is also acceptable. It is your call to choose the attribute or function-style form of the macro.

Reply:
Now this macro adopts function-like style.